src/MarkupConfig.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const YAML = require('yaml');
+const NamedMapProxy = require('./NamedMapProxy.js');
+const MarkupMappingSchema = require('./schemas/markupmapping.schema.json');
+const MarkupConfigValidator = require('./MarkupConfigValidator');
+
+class MarkupConfig {
+  constructor() {
+    this._source = '';
+    this._document = null;
+    this._markup = null;
+    this._cfg = {};
+  }
+
+  get markup() {
+    return this._markup;
+  }
+
+  withSource(value) {
+    this._source = value;
+    return this;
+  }
+
+  async validate() {
+    new MarkupConfigValidator().validate(this._cfg);
+  }
+
+  async loadConfig() {
+    if (this._source.indexOf('\t') >= 0) {
```

Comment:
Why not just `.trim()` the source to remove any extra whitespace at the start/end?

Reply:
?? this is to detect accidential tab characters. this has nothing to do with tirmming.

Reply:
`trim` takes care of tabs as well. Maybe I'm just missing some context to fully understand the issue here. Seems more a parsing issue to me than a config loading issue

Reply:
Tabs are not valid in YAML. Our YAML parser doesn’t reject them but interprets they as spaces or something similarly unexpected. To make things easy, we just reject YAML with tabs, as it has been biting us before.

Reply:
@trieloff ok. Then, just for the sake of separation of concern, I'd have expected to have that check as part of the parsing logic. Or even better, have a generic YAML parser util that takes care of it, instead of having to repeat that test potentially in several places of the helix project.

What happens if I fetch YAML as part of the pre.js, and forget to apply this extra check? Will every developer remember this extra check is required in 6 months from now?

Reply:
Let's burn that bridge when we cross it. 😉

In Helix, it's entirely fair to repeat yourself a bit (once), but once we have the same code thrice, it would be a good time to move it in a shared function.